Mathlib/Data/List/Count.lean

```diff
@@ -204,73 +236,85 @@ theorem count_append (a : α) : ∀ l₁ l₂, count a (l₁ ++ l₂) = count a
   countp_append _
 #align list.count_append List.count_append
 
-theorem count_join (l : List (List α)) (a : α) : l.join.count a = (l.map (count a)).Sum :=
+theorem count_join (l : List (List α)) (a : α) : l.join.count a = (l.map (count a)).sum :=
   countp_join _ _
 #align list.count_join List.count_join
 
-theorem count_concat (a : α) (l : List α) : count a (concat l a) = succ (count a l) := by
-  simp [-add_comm]
+theorem count_concat (a : α) (l : List α) : count a (concat l a) = succ (count a l) := by simp
 #align list.count_concat List.count_concat
 
 @[simp]
 theorem count_pos {a : α} {l : List α} : 0 < count a l ↔ a ∈ l := by
-  simp only [count, countp_pos, exists_prop, exists_eq_right']
+  simp only [count, countp_pos, beq_iff_eq, exists_eq_right]
 #align list.count_pos List.count_pos
 
 @[simp]
-theorem one_le_count_iff_mem {a : α} {l : List α} : 1 ≤ count a l ↔ a ∈ l :=
-  count_pos
+theorem one_le_count_iff_mem {a : α} {l : List α} : 1 ≤ count a l ↔ a ∈ l := count_pos
 #align list.one_le_count_iff_mem List.one_le_count_iff_mem
 
-@[simp]
+-- Porting note: lower priority to make simpNF linter happy
+@[simp 900]
 theorem count_eq_zero_of_not_mem {a : α} {l : List α} (h : a ∉ l) : count a l = 0 :=
   Decidable.by_contradiction fun h' => h <| count_pos.1 (Nat.pos_of_ne_zero h')
 #align list.count_eq_zero_of_not_mem List.count_eq_zero_of_not_mem
 
-theorem not_mem_of_count_eq_zero {a : α} {l : List α} (h : count a l = 0) : a ∉ l := fun h' =>
-  (count_pos.2 h').ne' h
+theorem not_mem_of_count_eq_zero {a : α} {l : List α} (h : count a l = 0) : a ∉ l :=
+  fun h' => (count_pos.2 h').ne' h
 #align list.not_mem_of_count_eq_zero List.not_mem_of_count_eq_zero
 
 @[simp]
-theorem count_eq_zero {a : α} {l} : count a l = 0 ↔ a ∉ l :=
+theorem count_eq_zero : count a l = 0 ↔ a ∉ l :=
   ⟨not_mem_of_count_eq_zero, count_eq_zero_of_not_mem⟩
 #align list.count_eq_zero List.count_eq_zero
 
 @[simp]
-theorem count_eq_length {a : α} {l} : count a l = l.length ↔ ∀ b ∈ l, a = b :=
-  countp_eq_length _
+theorem count_eq_length : count a l = l.length ↔ ∀ b ∈ l, a = b := by
+  rw [count, countp_eq_length]
+  refine ⟨fun h b hb => ?h₁, fun h b hb => ?h₂⟩
+  · refine' Eq.symm _
+    simpa using h b hb
+  · rw [h b hb, beq_self_eq_true]
 #align list.count_eq_length List.count_eq_length
 
 @[simp]
-theorem count_repeat (a : α) (n : ℕ) : count a (repeat a n) = n := by
-  rw [count, countp_eq_length_filter, filter_eq_self.2, length_repeat] <;>
-    exact fun b m => (eq_of_mem_repeat m).symm
-#align list.count_repeat List.count_repeat
+theorem count_replicate (a : α) (n : ℕ) : count a (replicate n a) = n := by
```

Comment:
For all lemmas about `repeat`, please keep the `repeat` version, but also add the `replicate` version. And mark the `repeat` version as deprecated. Since the order of arguments is changing it's good to keep both around, to make porting of future files smoother.